modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java

```diff
@@ -275,6 +276,30 @@ private ConfigurationException buildExclusiveParametersException(ParameterizedMo
                                                                  Joiner.on(", ").join(definedExclusiveParameters))));
   }
 
+  /**
+   * Gets a {@link ValueResolver} for the {@code parameterModel} if it has an associated a default value or encoding.
+   *
+   * @param parameterModel to be resolved
+   * @return a {@link ValueResolver} for cases such having a non null default value or depending on the runtime's default
+   *         encoding, {@code null} otherwise.
+   */
+  private ValueResolver<?> getValueResolverForAbsentParameter(ParameterModel parameterModel) {
```

Comment:
this method seems to overlap with the piece of code you added on resolveParameters(). Also the name of this method is misleading, since it's only for default encoding, not for any absent one